src/main/java/de/chojo/repbot/listener/voting/ReputationVoteListener.java

```diff
@@ -0,0 +1,118 @@
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.components.ActionRow;
+import net.dv8tion.jda.api.interactions.components.Button;
+import net.dv8tion.jda.api.interactions.components.ButtonStyle;
+import net.dv8tion.jda.api.interactions.components.Component;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ReputationVoteListener extends ListenerAdapter {
+    private static final Component DELETE = Button.of(ButtonStyle.DANGER, "vote:delete", Emoji.fromUnicode("🗑️"));
+    private static final Pattern VOTE = Pattern.compile("vote:(?<id>[0-9]*?)");
+    private final ReputationService reputationService;
+    private final ILocalizer loc;
+    private final Map<Long, VoteRequest> voteRequests = new HashMap<>();
+
+    public ReputationVoteListener(ReputationService reputationService, ILocalizer localizer) {
+        this.reputationService = reputationService;
+        this.loc = localizer;
+    }
+
+    @Override
+    public void onButtonClick(@NotNull ButtonClickEvent event) {
+        event.deferEdit().queue();
+        if (event.getButton() == null || event.getButton().getId() == null) return;
+
+        var matcher = VOTE.matcher(event.getButton().getId());
+        if (!matcher.find()) return;
+        if (!voteRequests.containsKey(event.getMessageIdLong())) return;
+        var voteRequest = voteRequests.get(event.getMessageIdLong());
+        if (!Verifier.equalSnowflake(voteRequest.member(), event.getMember())) return;
+        if ("vote:delete".equals(event.getButton().getId())) {
+            event.getMessage().delete().queue();
+            voteRequests.remove(event.getMessageIdLong());
+            return;
+        }
+
+        var target = voteRequest.getTarget(event.getButton().getId());
+
+        if (reputationService.submitReputation(event.getGuild(), event.getUser(), target.get().getUser(), voteRequest.refMessage(), null, ThankType.REACTION)) {
+            voteRequest.voted();
+            voteRequest.remove(event.getButton().getId());
+            voteRequest.voteMessage().
+                    editMessage(voteRequest.getNewEmbed(loc.localize("listener.messages.request.descrThank"
+                            , event.getGuild(), Replacement.create("MORE", voteRequest.remainingVotes()))))
+                    .setActionRows(getComponentRows(voteRequest.components()))
+                    .queue();
+            if (voteRequest.remainingVotes() == 0) {
+                voteRequest.voteMessage().delete().queueAfter(5, TimeUnit.SECONDS,
+                        suc -> voteRequests.remove(voteRequest.voteMessage().getIdLong()), err -> {
+                        });
```

Comment:
I assume you want to suppress `UNKNOWN_MESSAGE` here so pls make that explicit by using
`ErrorResponseException.ignore(ErrorResponse.UNKNOWN_MESSAGE)` as the error handler